### PR TITLE
Store cleanups

### PR DESCRIPTION
**Pull request by larskarlitski**
Some random cleanups of the store and its interaction with weldr, while I'm continuing to research towards #342.

Hope the commits are self-explanatory.

### CODE REVIEW

File: internal/store/store.go [hunk 1]
```diff
@@ -554,8 +554,10 @@ func (s *Store) getImageBuildDirectory(composeID uuid.UUID, imageBuildID int) st
 	return fmt.Sprintf("%s/%d", s.getComposeDirectory(composeID), imageBuildID)
 }
 
-func (s *Store) PushCompose(distro distro.Distro, arch distro.Arch, imageType distro.ImageType, composeID uuid.UUID, bp *blueprint.Blueprint, repos []rpmmd.RepoConfig, packages, buildPackages []rpmmd.PackageSpec, size uint64, uploadTarget *target.Target) error {
-	targets := []*target.Target{}
+func (s *Store) PushCompose(distro distro.Distro, arch distro.Arch, imageType distro.ImageType, composeID uuid.UUID, bp *blueprint.Blueprint, repos []rpmmd.RepoConfig, packages, buildPackages []rpmmd.PackageSpec, size uint64, targets []*target.Target) error {
+	if targets == nil {
```
Comment (teg) [inline internal/store/store.go]: This doesn't actually drop the appending of the local target, does it?

File: internal/weldr/api.go [hunk 2]
```diff
@@ -1411,8 +1411,25 @@ func (api *API) composeHandler(writer http.ResponseWriter, request *http.Request
 			statusResponseError(writer, http.StatusBadRequest, errors)
 			return
 		}
+		targets = append(targets, t)
 	}
 
+	filename, _, err := api.distro.FilenameFromType(cr.ComposeType)
```
Comment (teg) [inline internal/weldr/api.go]: `distro.FilenameFromType()` is deprecated. Simply use `imageType.Filename()` instead, and you can drop the error handling.

File: internal/weldr/api.go [hunk 3]
```diff
@@ -1411,8 +1411,25 @@ func (api *API) composeHandler(writer http.ResponseWriter, request *http.Request
...
+	if err != nil {
+		errors := responseError{
+			ID:  "BadCompose",
+			Msg: fmt.Sprintf("internal error: no filename for compose type %s", cr.ComposeType),
+		}
+		statusResponseError(writer, http.StatusBadRequest, errors)
+		return
+	}
+
+	targets = append(targets, target.NewLocalTarget(
```
Comment (teg) [inline internal/weldr/api.go]: Before this patch, this was skipped if the store did not have a state directory. I think that's the wrong place to have a conditional (do it when receiving the file, if anything), so this should be ok (that was only conditional for easier testing, so might never matter at all). But worth a comment in the commit message?

File: internal/store/store.go (hunk 1 above)
Comment (larskarlitski) [reply internal/store/store.go]: Oh dear, I missed that while rebasing on top of the refactor. Thanks for catching that!

File: internal/weldr/api.go (hunk 2 above)
Comment (larskarlitski) [reply internal/weldr/api.go]: Oh I get to drop the whole error case. Nice!

File: internal/weldr/api.go (hunk 3 above)
Comment (larskarlitski) [reply internal/weldr/api.go]: Good point. Added an explanation.

**Comment by larskarlitski** [comment]
> Love these cleanups! In addition to request to one of the patches, a `go fmt` is needed.

Thanks for the review! Should be good for another round.

**Comment by larskarlitski** [comment]
Update to resolve the conflict.

**Comment by larskarlitski** [comment]
This [passed on travis](https://travis-ci.org/github/larskarlitski/osbuild-composer/builds/666522179), but it failed to send status.

**Pull request merged**